pkgs/development/python-modules/testpath/default.nix

```diff
@@ -1,18 +1,22 @@
 { stdenv
+, lib
 , buildPythonPackage
 , fetchPypi
+, pythonOlder
+, pathlib2
 }:
 
 buildPythonPackage rec {
   pname = "testpath";
   version = "0.4.2";
-  format = "wheel";
 
   src = fetchPypi {
-    inherit pname version format;
-    sha256 = "46c89ebb683f473ffe2aab0ed9f12581d4d078308a3cb3765d79c6b2317b0109";
+    inherit pname version;
+    sha256 = "b694b3d9288dbd81685c5d2e7140b81365d46c29f5db4bc659de5aa6b98780f8";
   };
 
+  propagatedBuildInputs = lib.optional (pythonOlder "3.4") pathlib2;
```

Comment:
If using a python with the stdlib pathlib, it will just use that:
https://github.com/jupyter/testpath/blob/master/testpath/asserts.py#L4-L9

Otherwise it needs this.